### PR TITLE
Add functionality to edit session button and create a route for it

### LINKED ISSUE #1135: Add functionality to the edit buttons in the session route and create a route for the same
**I'm submitting a ...**  (check one with "x")
- [x] bug report
- [ ] feature request
- [ ] support request => Please do not submit support requests here, instead ask your query in out Gitter channel at https://gitter.im/fossasia/open-event-frontend

**Current behavior:**
Currently if the edit button is clicked it gets redirected to the route e/{event_identifier}/cfs/new-sessions which is a public route and has no specific session id in the route hence redirects to the first session of the route.

**Expected behavior:**
The route to edit a session should be - /events/{event_identifier}/sessions/edit/<session_id>

**Would you like to work on the issue?**

Yes

### PR DESCRIPTION
**Pull request by ritikamotwani**
#### Checklist

- [x] I have read the [Contribution & Best practices Guide](https://blog.fossasia.org/open-source-developer-guide-and-best-practices-at-fossasia).
- [x] My branch is up-to-date with the Upstream `development` branch.
- [x] The acceptance, integration, unit tests and linter pass locally with my changes 
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] I have added necessary documentation (if appropriate)

#### Short description of what this resolves:
The edit session button now redirects to a form to edit that particular session.

#### Changes proposed in this pull request:

- Form to edit a session is created.
- A route is created to edit the session: /events/{event_identifier}/view/sessions/edit/<session_id>
- Functionality to the edit session button is added.

Fixes #1135

### CODE REVIEW

**Comment by ritikamotwani** [comment]
@CosmicCoder96 did the changes please review.
@geekyd @niranjan94 @pradeepgangwar @mayank8318 please review.

File: app/templates/events/view/sessions.hbs [hunk 1]
```diff
@@ -1,6 +1,6 @@
 {{#if model.isSessionsSpeakersEnabled}}
   <div class="ui grid stackable">
-    {{#if (not-eq session.currentRouteName 'events.view.sessions.create')}}
+    {{#if (and (not-eq session.currentRouteName 'events.view.sessions.create') (not-eq session.currentRouteName 'events.view.sessions.edit'))}}
```
Comment (niranjan94) [inline app/templates/events/view/sessions.hbs]: Try to reduce the complexity of conditionals in templates. handlebars was designed to keep as less of logic in the template as possible. Move this to a computed property on the controller for better readability.

File: app/controllers/events/view/sessions.js [hunk 2]
```diff
@@ -0,0 +1,12 @@
+import Controller from '@ember/controller';
+import { computed } from '@ember/object';
+
+export default Controller.extend({
+  currentRouteName: computed('session.currentRouteName', function() {
```
Comment (CosmicCoder96) [inline app/controllers/events/view/sessions.js]: this computed property returns a boolean. The name `currentRouteName` is counter-intuitive for that.

**Comment by ritikamotwani** [comment]
@CosmicCoder96 did the changes.

**Comment by ritikamotwani** [comment]
@geekyd @niranjan94 please review.

File: app/routes/events/view/sessions/edit.js [hunk 3]
```diff
@@ -0,0 +1,13 @@
+import Route from '@ember/routing/route';
+import AuthenticatedRouteMixin from 'ember-simple-auth/mixins/authenticated-route-mixin';
+
+export default Route.extend(AuthenticatedRouteMixin, {
+  titleToken() {
+    return this.get('l10n').t('Edit Sessions');
```
Comment (niranjan94) [inline app/routes/events/view/sessions/edit.js]: Title should be like `<session name> - Edit - <event name>`

**Comment by ritikamotwani** [comment]
@niranjan94 please review.

**Pull request merged**